sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/AvroCarbonWriter.java

```diff
@@ -823,6 +829,31 @@ public void write(Object object) throws IOException {
     }
   }
 
+  /**
+   * Load data of all avro files at given location iteratively.
+   *
+   * @throws IOException
+   */
+  @Override
+  public void write() throws IOException {
+    if (this.dataFiles == null || this.dataFiles.length == 0) {
+      throw new RuntimeException("'withAvroPath()' must be called to support loading avro files");
+    }
+    Arrays.sort(this.dataFiles, Comparator.comparing(CarbonFile::getPath));
+    for (CarbonFile dataFile : this.dataFiles) {
+      this.loadSingleFile(dataFile);
+    }
+  }
+
+  private void loadSingleFile(CarbonFile file) throws IOException {
+    DataFileStream<GenericData.Record> avroReader = SDKUtil
+        .buildAvroReader(file, this.configuration);
```

Comment:
`avroReader` stream is not closed neither in success case not in failure/exception cases. File InputStream used to create this DataFileStream in buildAvroReader is not closed. Check for all the stream reader/writer cases in this PR.

Reply:
Closed now all the stream reader.